custom_components/tahoma/sensor.py

```diff
@@ -1,116 +1,338 @@
-}
+
+@dataclass
+class OverkizSensorDescription(SensorEntityDescription):
+    """Class to describe a Overkiz sensor."""
+
+    value: Callable[[Any], Any] | None = lambda val: val
+
+
+SENSOR_DESCRIPTIONS = [
+    OverkizSensorDescription(
+        key="core:BatteryLevelState",
+        name="Battery Level",
+        unit_of_measurement=PERCENTAGE,
+        device_class=sensor.DEVICE_CLASS_BATTERY,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="core:BatteryState",
+        name="Battery",
+        device_class=sensor.DEVICE_CLASS_BATTERY,
+        value=lambda value: str(value).capitalize(),
+    ),
+    OverkizSensorDescription(
+        key="core:RSSILevelState",
+        name="RSSI Level",
+        value=lambda value: round(value),
+        unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        device_class=sensor.DEVICE_CLASS_SIGNAL_STRENGTH,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="core:ExpectedNumberOfShowerState",
+        name="Expected Number Of Shower",
+        icon="mdi:shower-head",
+        value=lambda value: round(value),
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="core:NumberOfShowerRemainingState",
+        name="Number of Shower Remaining",
+        icon="mdi:shower-head",
+        value=lambda value: round(value),
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    # V40 is measured in litres (L) and shows the amount of warm (mixed) water with a temperature of 40 C, which can be drained from a switched off electric water heater.
+    OverkizSensorDescription(
+        key="core:V40WaterVolumeEstimationState",
+        name="Water Volume Estimation at 40 °C",
+        icon="mdi:water",
+        value=lambda value: round(value),
+        unit_of_measurement=VOLUME_LITERS,
+        entity_registry_enabled_default=False,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="core:WaterConsumptionState",
+        name="Water Consumption",
+        icon="mdi:water",
+        value=lambda value: round(value),
+        unit_of_measurement=VOLUME_LITERS,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="io:OutletEngineState",
+        name="Outlet Engine",
+        icon="mdi:fan-chevron-down",
+        value=lambda value: round(value),
+        unit_of_measurement=VOLUME_LITERS,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="io:InletEngineState",
+        name="Inlet Engine",
+        icon="mdi:fan-chevron-up",
+        value=lambda value: round(value),
+        unit_of_measurement=VOLUME_FLOW_RATE_CUBIC_METERS_PER_HOUR,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="hlrrwifi:RoomTemperatureState",
+        name="Room Temperature",
```

Comment:
To ease the code, we can generate the name from the key.

Reply:
I thought about this as well, however I don't see the value. This lookup is fast and gives us a way to easily change the name.

We can always add this in a follow up PR? However, I think there are more pressing issues to solve :-).

Reply:
Sure. It’s just I like to reduce number of lines :D